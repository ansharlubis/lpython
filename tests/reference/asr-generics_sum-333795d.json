{
    "basename": "asr-generics_sum-333795d",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/generics_sum.py",
    "infile_hash": "2bc10020aa717dd902d2bc3c51b6b7f1a07ed66b1738134ac23b3543",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-generics_sum-333795d.stdout",
    "stdout_hash": "81cf6c32ec51bea1d310652e91d8d842169100ec281969320057a2a7",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}