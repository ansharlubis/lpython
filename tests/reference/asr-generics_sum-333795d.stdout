(TranslationUnit (SymbolTable 1 {T: (Variable 1 T Local () () Default (TypeParameter T) Source Public Required .false.), __lpython_generic_sum_0: (Function (SymbolTable 3 {_lpython_return_variable: (Variable 3 _lpython_return_variable ReturnVar () () Default (Integer 4 []) Source Public Required .false.), i: (Variable 3 i Local () () Default (Integer 4 []) Source Public Required .false.), l: (Variable 3 l Local () () Default (Integer 4 []) Source Public Required .false.), lst: (Variable 3 lst In () () Default (List (Integer 4 [])) Source Public Required .false.), res: (Variable 3 res Local () () Default (Integer 4 []) Source Public Required .false.)}) __lpython_generic_sum_0 [(Var 3 lst)] [(= (Var 3 l) (ListLen (Var 3 lst) (Integer 4 []) ()) ()) (= (Var 3 res) (IntegerBinOp (Var 3 res) Add (Var 3 i) (Integer 4 []) ()) ()) (= (Var 3 _lpython_return_variable) (Var 3 res) ()) (Return)] (Var 3 _lpython_return_variable) Source Public Implementation ()), _lpython_main_program: (Subroutine (SymbolTable 5 {}) _lpython_main_program [] [(Print () [(FunctionCall 1 __lpython_generic_sum_0 () [((ListConstant [(IntegerConstant 1 (Integer 4 [])) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 3 (Integer 4 []))] (List (Integer 4 []))))] (Integer 4 []) () ())] () ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 4 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())]), sum: (TemplateFunction (SymbolTable 2 {_lpython_return_variable: (Variable 2 _lpython_return_variable ReturnVar () () Default (TypeParameter T) Source Public Required .false.), i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), l: (Variable 2 l Local () () Default (Integer 4 []) Source Public Required .false.), lst: (Variable 2 lst In () () Default (List (TypeParameter T)) Source Public Required .false.), res: (Variable 2 res Local () () Default (TypeParameter T) Source Public Required .false.)}) sum [(Var 2 lst)] [(= (Var 2 l) (ListLen (Var 2 lst) (Integer 4 []) ()) ()) (DoLoop ((Var 2 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (Var 2 l) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(= (Var 2 res) (TemplateBinOp (Var 2 res) Add (ArrayItem (Var 2 l) [(() (Var 2 i) ())] (Integer 4 []) ()) (TypeParameter T) ()) ())]) (= (Var 2 _lpython_return_variable) (Var 2 res) ()) (Return)] (Var 2 _lpython_return_variable) Source Public Implementation ())}) [])
